common/src/main/scala/workers/WorkerClient.scala

```diff
@@ -76,4 +96,12 @@ class WorkerClient[F[_]: Concurrent: UUIDGen: Logger, R: Pickler](worker: WebWor
     Pickler[requestMessage.ResponseType]
   ): F[Unit] =
     request(requestMessage).use(_.compile.drain)
-}
+
+object WorkerClient:
+  def fromWorker[F[_]: Concurrent: UUIDGen: Logger, R: Pickler](
+    worker: WebWorkerF[F]
+  ): Resource[F, WorkerClient[F, R]] =
+    for {
+      latch  <- Resource.eval(Deferred[F, Unit])
```

Comment:
What are we using the latch for?

Reply:
The latch is released when the `Ready` message is received from the server. This allows us to wait asynchronously for the server to report that it's up. In line 50 we wait for the latch before sending a request.

I tried the `WorkerClient` initialization to wait synchronously for the server but explore took noticeably longer to start up.